src/pages/data.test.js

```diff
@@ -0,0 +1,427 @@
+import '@testing-library/jest-dom'
+import { CustomDataProvider } from '@dhis2/app-runtime'
+import { configure, render, waitFor, within } from '@testing-library/react'
+import React from 'react'
+import { MemoryRouter, Route } from 'react-router-dom'
+import { QueryParamProvider } from 'use-query-params'
+import { ReactRouter6Adapter } from 'use-query-params/adapters/react-router-6'
+import { formatData } from '../components/view/data-workspace/requests-display/index.js'
+import { getRelativeTimeDifference } from '../components/view/data-workspace/title-bar/index.js'
+import { getReportText } from '../components/view/submit-modal/submit-modal.js'
+import { AppContext, UserContext } from '../context/index.js'
+import {
+    testDataExchange,
+    testDataExchangeSourceData,
+    testImportSummary,
+    testRequest,
+    testUserContext,
+} from '../utils/builders.js'
+import { DataPage } from './data.js'
+
+const lastAnalyticsTableSuccess = '2024-07-07T21:47:58.383'
+const serverDate = '2024-07-18T17:36:38.164'
+const contextPath = 'debug.dhis2.org/dev'
+let mockImportSummariesResponse = {}
```

Comment:
is it possible to mockReturnValue at setup for useDataEngine, so that we don't store the importSummaryReponse in a global variable and then also clear mocks after each test? It seems like we'll be carrying over the mocked import summary from the previous tests here? (I might be misunderstanding this)